src/main/plugins/org.dita.pdf2/xsl/common/topicmergeImpl.xsl

```diff
@@ -46,26 +46,108 @@ See the accompanying LICENSE file for applicable license.
           <xsl:copy-of select="$map/@*"/>
           <xsl:apply-templates select="$map" mode="build-tree"/>
         </xsl:element>
     </xsl:template>
+    
+    <xsl:template match="*[contains(@class,' map/topicref ')][@first_topic_id]" mode="resolve-root-dita">
+        <xsl:param name="ditaDocs" as="element()+"/>
+        <xsl:variable name="topicId" select="substring-after(@first_topic_id,'#')"/>
+        <xsl:choose>
+            <xsl:when test="$ditaDocs/*[@id=$topicId]">
+                <xsl:variable name="topicref" select="." as="element()"/>
+                <xsl:for-each select="$ditaDocs/*[@id=$topicId]/*[contains(@class,' topic/topic ')]">
+                    <xsl:variable name="updateTopicTarget" select="concat('#',@id)"/>
+                    <xsl:variable name="lastTopic" select="if (position()=last()) then true() else false()" as="xs:boolean"/>
+                    <xsl:for-each select="$topicref">
+                        <xsl:copy>
+                            <xsl:copy-of select="@*[not(name()='href' or name()='first_topic_id')]"/>
+                            <xsl:attribute name="first_topic_id"><xsl:value-of select="$updateTopicTarget"/></xsl:attribute>
+                            <xsl:attribute name="href"><xsl:value-of select="$updateTopicTarget"/></xsl:attribute>
+                            <xsl:choose>
+                                <xsl:when test="$lastTopic">
+                                    <xsl:apply-templates select="*|comment()|processing-instruction()" mode="resolve-root-dita">
+                                        <xsl:with-param name="ditaDocs" select="$ditaDocs" as="element()+"/>
+                                    </xsl:apply-templates>
+                                </xsl:when>
+                                <xsl:otherwise>
+                                    <xsl:sequence select="*[contains(@class,' map/topicmeta ')]"/>
+                                </xsl:otherwise>
+                            </xsl:choose>
+                        </xsl:copy>
+                    </xsl:for-each>
+                </xsl:for-each>
+            </xsl:when>
+            <xsl:otherwise>
+                <xsl:copy>
+                    <xsl:apply-templates select="@*|*|text()|comment()|processing-instruction()" mode="resolve-root-dita">
+                        <xsl:with-param name="ditaDocs" select="$ditaDocs" as="element()+"/>
+                    </xsl:apply-templates>
+                </xsl:copy>
+            </xsl:otherwise>
+        </xsl:choose>
+    </xsl:template>
+    
+    <xsl:template match="@href[starts-with(.,'#')]" mode="resolve-root-dita">
+        <xsl:param name="ditaDocs" as="element()+"/>
+        <xsl:variable name="topicId" select="substring-after(.,'#')"/>
+        <xsl:choose>
+            <xsl:when test="$ditaDocs/*[@id=$topicId]">
+                <xsl:attribute name="href">
+                    <xsl:value-of select="concat('#',$ditaDocs/*[@id=$topicId]/*[1]/@id)"/>
```

Comment:
Set value of attribute directly with `xsl:attribute/@select` instead of nested `xsl:value-of`